doc/ci_admin_guide/Docker_registry_provisioning.md

```diff
@@ -0,0 +1,152 @@
+# Provisioning Docker registry
+
+## Preparing machine
+
+Spawn a VM using the CentOS / Ubuntu image (the rest of instruction will assume (CentOS 7),
+use DHCP and name `winci-registry`
+
+Log in using ssh and set the hostname in `/etc/hostname`.
+
+Update the system using `yum upgrade` and reboot.
+
+### Networking
+
+Log in using ssh and change the IP configuration to manual, by modifying the following file:
+`/etc/sysconfig/network-scripts/ifcfg-ens192`. Change the line with `BOOTPROTO` to
+
+```ini
+BOOTPROTO=static
+```
+
+and add the following lines:
+
+```ini
+IPADDR=10.84.12.27
+NETMASK=255.255.255.0
+GATEWAY=10.84.12.254
+DNS1=172.21.200.60
+DNS2=172.29.131.60
+DNS3=8.8.8.8
+DOMAIN="contrail.juniper.net englab.juniper.net juniper.net jnpr.net"
+```
+
+Then reboot the machine and log in using the new IP.
+
+## Installing docker
+
+```bash
+yum install docker
+systemctl enable docker
+systemctl start docker
+```
+
+## Configuring registry
+
+### Pulling images
+
+First, we need to add `ci-repo` to insecure registries, as it uses HTTP:
+
+Add this line to `/etc/docker/daemon.json` (if the file is not there, create it):
+
+```json
+{ "insecure-registries":["ci-repo.englab.juniper.net:5000"] }{}
+```
+
+Restart docker:
+```bash
+systemctl restart docker
+```
+
+Then, we need to determine a list of images to pull. You can run this command
+on any working controller (*not on a registry machine!*):
```

Comment:
What is a controller?

Also, in generall, we should weird out-of-nowhere comments in parantheses in technical documentation.

Reply:
Actually, I really don't like this step of logging to yet another machine to get a list of images to copy... But it's the most reliable way to determine which images are needed in our CI. Not sure what to do about it.